CMakePresets.json

```diff
@@ -0,0 +1,71 @@
+{
+    "version": 6,
+    "cmakeMinimumRequired": {
+        "major": 3,
+        "minor": 14,
+        "patch": 0
+      },
+    "configurePresets": [
+        {
+            "name": "common",
+            "hidden": true,
+            "cacheVariables": {
+                "FREECAD_USE_PYBIND11:BOOL": "ON"
+            }
+        },
+        {
+            "name": "debug",
+            "displayName": "Debug",
+            "description": "Default debug profile",
+            "binaryDir": "${sourceDir}/build/debug",
+            "inherits": [
+                "common"
+            ],
+            "cacheVariables": {
+                "CMAKE_BUILD_TYPE": "Debug"
+            }
+        },
+        {
+            "name": "release",
+            "displayName": "Release",
+            "description": "Default release profile",
+            "binaryDir": "${sourceDir}/build/release",
+            "inherits": [
+                "common"
+            ],
+            "cacheVariables": {
+                "CMAKE_BUILD_TYPE": "Release"
+            }
+        },
+        {
+            "name": "conda",
+            "hidden": true,
+            "cmakeExecutable": "${workspaceFolder}/.conda/freecad/bin/cmake",
+            "cacheVariables": {
+                "BUILD_WITH_CONDA:BOOL": "ON",
```

Comment:
The format of this file requires the following definitions:

                "BUILD_WITH_CONDA": {
                    "type": "BOOL",
                    "value": "ON"
                },
                "FREECAD_USE_EXTERNAL_SMESH": {
                    "type": "BOOL",
                    "value": "ON"
                }